utils/video.py

```
@@ -0,0 +1,66 @@
+        str: Path to downloaded temporary file
+        
+    Raises:
+        Exception: If download fails
+    """
+    temp_file = tempfile.NamedTemporaryFile(suffix='.mp4', delete=False)
+    output_path = temp_file.name
+    temp_file.close()
+    
+    print(f"Downloading video from: {url}")
+    try:
+        urllib.request.urlretrieve(url, output_path)
```

Ruff (0.13.3)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)